File: lib/models/news_model.dart
class News {
  final String title;
  final String description;
  final String content;
  final String imageUrl;
  final String category;
  final DateTime publishedAt;
  final String source;

  News({
    required this.title,
    required this.description,
    required this.content,
    required this.imageUrl,
    required this.category,
    required this.publishedAt,
    required this.source,
  });

  factory News.fromJson(Map<String, dynamic> json) {
    return News(
      title: json['title'] ?? 'No Title',
      description: json['description'] ?? 'No Description',
      content: json['content'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
      category: json['category'] ?? 'General',
      publishedAt: DateTime.parse(
        json['publishedAt'] ?? DateTime.now().toIso8601String(),
      ),
      source: json['source'] ?? 'Unknown Source',
    );
  }
}

File: api/config/models/news_model.js
const mongoose = require('mongoose');

const newsScheme = new mongoose.Schema({
    title: {
        type: String,
        required: [true, 'Please add a title'],
    },

    description: {
        type: String,
    },

    content: {
        type: String,
    },

    imageUrl: {
        type: String,
    },

    category: {
        type: String,
        enum: ['Technology', 'Health', 'Sports', 'Business', 'Entertainment', 'Science', 'World', 'Politics', 'Travel', 'Lifestyle'],
        required: [true, 'Please add a category']
    },

    publishedAt: {
        type: Date,
        default: Date.now,
    },

    source: {
        type: String,
    },

}, { timestamps: true });

const News = mongoose.model('News', newsScheme);
module.exports = News;